src/System.Private.CoreLib/shared/System/Collections/Generic/Comparer.cs

```diff
@@ -10,21 +10,18 @@
 namespace System.Collections.Generic
 {
     [Serializable]
-    [TypeDependencyAttribute("System.Collections.Generic.ObjectComparer`1")]
     [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")] 
-    public abstract class Comparer<T> : IComparer, IComparer<T>
+    public abstract partial class Comparer<T> : IComparer, IComparer<T>
```

Comment:
> System.Runtime.CompilerServices.JitHelpers that don't exist outside CoreCLR

We will need to do something about this one.

Reply:
It does already exist in Mono though

Reply:
It seems that both in CoreCLR and CoreRT it's an intrinsic (even though in CoreCLR it lacks the `[Intrinsic]` attribute). Maybe it would make sense to unify the name to something the is not JIT specific like `EqualityComparerHelpers.EnumOnlyEquals` in the CoreRT. @jkotas, thoughts?

https://github.com/dotnet/corert/blob/f0bcd9c2dca8b4fc2cb7de9b6ae80dbf8d584229/src/System.Private.CoreLib/src/Internal/IntrinsicSupport/EqualityComparerHelpers.cs#L140-L144
https://github.com/dotnet/coreclr/blob/a548a0e6d1aa8ee795f024f129d5393c02328370/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/jithelpers.cs#L88-L93
https://github.com/dotnet/coreclr/blob/a548a0e6d1aa8ee795f024f129d5393c02328370/src/vm/jitinterface.cpp#L6996-L7087

Reply:
Mono has `JitHelpers`, but it lacks `EnumEquals` member. The old reference source implementation had a different method called `UnsafeEnumCast`. Thus whatever name is used it would have to be added to Mono anyway.

Reply:
Yes, something like this would help.